scapy/interfaces.py

```diff
@@ -0,0 +1,347 @@
+# This file is part of Scapy
+# See http://www.secdev.org/projects/scapy for more information
+# Copyright (C) Philippe Biondi <phil@secdev.org>
+# Copyright (C) Gabriel Potter <gabriel@potter.fr>
+# This program is published under a GPLv2 license
+
+"""
+Interfaces management
+"""
+
+import itertools
+import uuid
+from collections import defaultdict
+
+from scapy.config import conf
+from scapy.consts import WINDOWS
+from scapy.utils import pretty_list
+
+from scapy.modules.six.moves import UserDict
+import scapy.modules.six as six
+
+
+class InterfaceProvider(object):
+    name = "Unknown"
+    headers = ("INDEX", "IFACE", "IPv4", "IPv6", "MAC")
+    header_sort = 3
+    libpcap = False
+
+    def load(self):
+        """Returns a dictionary of the loaded interfaces, by their
+        name."""
+        raise Exception("load() should be implemented !")
+
+    def reload(self):
+        """Same than load() but for reloads. By default calls load"""
+        return self.load()
+
+    def l2socket(self):
+        """Return L2 socket used by interfaces of this provider"""
+        return conf.L2socket
+
+    def l2listen(self):
+        """Return L2listen socket used by interfaces of this provider"""
+        return conf.L2listen
+
+    def l3socket(self):
+        """Return L3 socket used by interfaces of this provider"""
+        return conf.L3socket
+
+    def _is_valid(self, dev):
+        """Returns whether an interface is valid or not"""
+        return dev.valid
+
+    def _format(self, dev, **kwargs):
+        """Returns a tuple of the elements used by show()"""
+        mac = dev.mac
+        resolve_mac = kwargs.get("resolve_mac", True)
+        if resolve_mac and conf.manufdb:
+            mac = conf.manufdb._resolve_MAC(mac)
+        index = str(dev.index)
+        return (index, dev.description, str(dev.ip), str(dev.ip6), mac)
```

Comment:
I think that we need to display all addresses, or find a good compromise. Currently, for IPv6, this only displays link-local addresses (fe80::/64) which is a bit confusing.